src/Akeneo/Connectivity/Connection/back/Application/Webhook/WebhookEventBuilder.php

```diff
@@ -27,12 +27,17 @@ public function __construct(iterable $builders)
         $this->builders = $builders;
     }
 
-    public function build(BusinessEventInterface $businessEvent): WebhookEvent
+    /**
+     * @param array<mixed> $context
+     */
+    public function build(BusinessEventInterface $businessEvent, array $context = []): WebhookEvent
     {
         return new WebhookEvent(
             $businessEvent->name(),
             $businessEvent->uuid(),
             date(\DateTimeInterface::ATOM, $businessEvent->timestamp()),
+            $businessEvent->author(),
+            array_key_exists('pim_source', $context) ? $context['pim_source'] : '',
```

Comment:
You can use the null coalescing operator in this case :)
```suggestion
            $context['pim_source'] ?? ''
```
I wonder if we should not raise an error if 'pim_source' is not defined instead?!
Like `InvalidArgumentException('Context property 'pim_source' is mandatory');`
It's probably better to break early than having a "silent" error

Reply:
After our discussion, I added the Exception. See https://github.com/akeneo/pim-community-dev/pull/12891/commits/f6a8023ef310b282b3af774edf0f19e4d21ee013